uint32 TYPE_UINT8=0
uint32 TYPE_UINT16=1
uint32 TYPE_UINT32=2
uint32 TYPE_UINT64=3
uint32 TYPE_INT8=4
uint32 TYPE_INT16=5
uint32 TYPE_INT32=6
uint32 TYPE_INT64=7
uint32 TYPE_FLOAT=8
uint32 TYPE_DOUBLE=9
uint32 TYPE_BOOL=10
uint32 TYPE_STRING=11

string name
uint32 value_type

uint8[] uint8_seq
uint16[] uint16_seq
uint32[] uint32_seq
uint64[] uint64_seq
int8[] int8_seq
int16[] int16_seq
int32[] int32_seq
int64[] int64_seq
float32[] float32_seq
float64[] float64_seq
bool[] bool_seq
string[] string_seq
